jadx-gui/src/main/java/jadx/gui/ui/dialog/RenameDialog.java

```diff
@@ -69,7 +69,12 @@ private RenameDialog(MainWindow mainWindow, JRenameNode node) {
 	}
 
 	private void initRenameField() {
-		renameField.setText(node.getName());
+		if (node instanceof JRenamePackage) {
+			JRenamePackage renamePackage = (JRenamePackage) node;
+			renameField.setText(renamePackage.getTitle());
```

Comment:
Well, if you want to see full name instead short name in editor box, you can just change "name" to "fullName" in `JRenamePackage.getName()` method.
But, short name used here intentionally, because detecting changed part in full name is much harder. It is unclear if the middle package name was renamed or rest moved into a new package. Also, packages count can change and that complicate things even more.
I am actually implemented a full package rename, but it is just safer and easier to rename only one part.

I will also look into this issue, maybe I will be able to find a root cause.

Reply:
> I will also look into this issue, maybe I will be able to find a root cause.

I thik here root problem
https://github.com/skylot/jadx/blob/a80eefaeea3be4eeb5e1ff274549a5c058bfbcd3/jadx-gui/src/main/java/jadx/gui/utils/pkgs/PackageHelper.java#L64